### PR TITLE
remove `sp-serializer` which is not useful

### PR DESCRIPTION
**Pull request by yjhmelody**
I think this is a legacy crate which is not very helpful anymore.

Cumulus Companion: https://github.com/paritytech/cumulus/pull/2574

--- 
polkadot address: 15ouFh2SHpGbHtDPsJ6cXQfes9Cx1gEFnJJsJVqPGzBSTudr

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2828741

**Review by davxy — APPROVED** [review]

File: primitives/core/Cargo.toml [hunk 1]
```diff
@@ -56,7 +56,7 @@ sp-runtime-interface = { version = "7.0.0", default-features = false, path = "..
 w3f-bls = { version = "0.1.3", default-features = false, optional = true}
 
 [dev-dependencies]
-sp-serializer = { version = "4.0.0-dev", path = "../serializer" }
+serde = { version = "1.0.136" }
```
Comment (bkchr) [inline primitives/core/Cargo.toml:59]:
```suggestion
serde = "1.0.136"
```

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2830018

**Comment by ggwpez** [comment]
bot fmt

**Review by ggwpez — APPROVED** [review]

**Comment by ggwpez** [comment]
bot merge

**Comment by ggwpez** [comment]
It is used in cumulus. Could you do a companion please?

**Comment by yjhmelody** [comment]
@ggwpez ok

**Comment by ggwpez** [comment]
bot merge

**Review by michalkucharczyk — APPROVED** [review]

**Review by mrcnski — APPROVED** [review]

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**